airflow-core/src/airflow/traces/tracer.py

```diff
@@ -276,29 +283,45 @@ def __call__(cls, *args, **kwargs):
             cls._initialize_instance()
         return cls.instance
 
-    @classmethod
     def configure_factory(cls):
         """Configure the trace factory based on settings."""
-        if conf.has_option("traces", "otel_on") and conf.getboolean("traces", "otel_on"):
+        otel_on = conf.getboolean("traces", "otel_on")
+
+        if cls.check_debug_traces_flag:
+            debug_traces_on = conf.getboolean("traces", "otel_debug_traces_on")
+        else:
+            # Set to true so that it will be ignored during the evaluation for the factory instance.
+            debug_traces_on = True
```

Comment:
Could you explain the logic of this True here - I am not sure i follow why debugging should default to True here?

Reply:
`debug_traces_on` is only used in line 296

```python
if otel_on and debug_traces_on:
```

if we don't want to check that flag
```python
if cls.check_debug_traces_flag:
```

then by setting it to `True` it's always ignored in the next evaluation. The condition will always evaluate to whatever value `otel_on` has.

* if `otel_on` is true and `debug_traces_on` is true, then it will be true
* if `otel_on` is false and `debug_traces_on` is true, then it will be false

Reply:
I extended the comment to explain it better.